### PR TITLE
Add test cases for JSON resource marshaling

### PR DESCRIPTION
**Pull request by sagar23sj**
Resource Covered :

* InstallationRepositoriesEvent
* TeamEvent
* StarEvent
* TeamAddEvent

Helps with issue: #55

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @sagar23sj !
LGTM, although we might want to wait on merging this one until #1900 is merged... as that one will definitely break this one.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
@sagar23sj - either you can resolve the conflicts or I can - either way is fine with me, please just let me know.

**Pull request closed without merging**
